packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -0,0 +1,216 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, named } from 'inversify';
+import ElectronStorage = require('electron-store');
+import nativeKeymap = require('native-keymap');
+import { ContributionProvider, MaybePromise } from '../common';
+
+import * as electron from 'electron';
+import { fork } from 'child_process';
+
+export const ElectronMainApplicationContribution = Symbol('ElectronMainApplicationContribution');
+export interface ElectronMainApplicationContribution {
+
+    start(app: Electron.App): MaybePromise<void>;
```

Comment:
lifecycle callbacks should start with `on`, i would prefer if we follow events naming for them: https://github.com/theia-ide/theia/wiki/Coding-Guidelines#names

i.e. `onDidStart` or `onWillStart` depends when it's called

lifecycle callbacks should be optional as well

Reply:
Makes sense indeed, will change